nucypher/cli/actions/migrate.py

```diff
@@ -0,0 +1,44 @@
+from pathlib import Path
+
+import click
+
+from nucypher.config.migrations import MIGRATIONS
+from nucypher.config.migrations.common import (
+    InvalidMigration,
+    WrongConfigurationVersion,
+)
+from nucypher.utilities.emitters import StdoutEmitter
+
+
+def migrate(emitter: StdoutEmitter, config_file: Path):
+    for jump, migration in MIGRATIONS.items():
+        old, new = jump
+        emitter.message(f"Checking migration {old} -> {new}")
+        if not migration:
+            emitter.echo(
+                f"Migration {old} -> {new} not found.",
+                color="yellow",
+                verbosity=1,
+            )
+            continue  # no migration script
```

Comment:
> Do we assume that not having a migration is ok (because it may not be needed) or are we going to throw an exception somewhere else?

Fair question - essentially both. `no-ops` shouldn't really happen because then what was the point of updating the configuration version. There is no migration for v2->v3; I think (?) it was because we didn't do the migration in conjunction with it at the time, and we may have lost track of what we changed 😞 , but that was a long time ago. 

The migration is done in substeps - https://github.com/nucypher/nucypher/blob/v7.2.x/nucypher/config/migrations/__init__.py. If the migration fails at any point in that chain, then an exception is raised. In any case, if any single step of migration fails, the entire thing fails anyway. So the fact that v2->v3 is a no-op means that **if** we started with a v1 or v2 file and the resulting v3 file can't be migrated to v4 then the whole migration fails. If it can be migrated from v3 -> v4 then no problem and we continue with the remaining migrations to latest.

That being said I don't believe there are any actual config versions lower than v4 (`nucypher v6.0.0`) out there - we can handle migrations from v4 all the way to latest - there's a test for this, https://github.com/nucypher/nucypher/blob/v7.2.x/tests/integration/config/test_configuration_migrations.py#L42.

Reply:
What do you think about the alternative, i.e. filling the missing migrations with `no-ops` and throwing when no migration is found?